src/library_ccall.js

```diff
@@ -33,9 +33,7 @@ mergeInto(LibraryManager.library, {
         var ret = 0;
         if (str !== null && str !== undefined && str !== 0) { // null string
           // at most 4 bytes per UTF-8 code point, +1 for the trailing '\0'
-          var len = (str.length << 2) + 1;
-          ret = stackAlloc(len);
-          stringToUTF8(str, ret, len);
+          ret = allocateUTF8OnStack(str);
```

Comment:
This looks very fishy: it both returns a pointer to stack memory and it does a stack allocation without ever restoring the stack pointer.

Reply:
But that is exactly what the `stackAlloc` function also does, right?   This is just like a `stackAlloc` with some extra copying, right?

Reply:
Fixes.

Reply:
Oh I see, this is used in a safe way below. Sounds good.